# Project 3 -- SEYI OGUNMODEDE

**INSTRUCTOR:** Dr. WARD

- Help with video in figuring out how to write a function.
    

    


## Question 1

In [2]:
# ls - it means file listing , to show all the files in the folder
ls /anvil/projects/tdm/data/flights/subset/

1987.csv*     1994.csv*     2001.csv*     2008.csv*
1987.parquet  1994.parquet  2001.parquet  2008.parquet
1988.csv*     1995.csv*     2002.csv*     airports.csv*
1988.parquet  1995.parquet  2002.parquet  airports.parquet
1989.csv*     1996.csv*     2003.csv*     carriers.csv*
1989.parquet  1996.parquet  2003.parquet  carriers.parquet
1990.csv*     1997.csv*     2004.csv*     plane-data.csv*
1990.parquet  1997.parquet  2004.parquet  plane-data.parquet
1991.csv*     1998.csv*     2005.csv*     sampler.sh*
1991.parquet  1998.parquet  2005.parquet  variable-descriptions.csv*
1992.csv*     1999.csv*     2006.csv*     variable-descriptions.parquet
1992.parquet  1999.parquet  2006.parquet
1993.csv*     2000.csv*     2007.csv*
1993.parquet  2000.parquet  2007.parquet


In [4]:
ls /anvil/projects/tdm/data/flights/subset/2003.csv

/anvil/projects/tdm/data/flights/subset/2003.csv*


In [6]:
# for more information about the files

ls -la /anvil/projects/tdm/data/flights/subset/2003.csv

-rwxrwxr-x 1 7095091 7500002 626745242 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/2003.csv*


In [9]:
# to check all of the files that begins with 2

ls -la /anvil/projects/tdm/data/flights/subset/2*.csv

-rwxrwxr-x 1 7095091 7500002 570151613 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/2000.csv*
-rwxrwxr-x 1 7095091 7500002 596910261 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/2001.csv*
-rwxrwxr-x 1 7095091 7500002 530036430 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/2002.csv*
-rwxrwxr-x 1 7095091 7500002 626745242 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/2003.csv*
-rwxrwxr-x 1 7095091 7500002 669879113 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/2004.csv*
-rwxrwxr-x 1 7095091 7500002 671027265 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/2005.csv*
-rwxrwxr-x 1 7095091 7500002 672068096 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/2006.csv*
-rwxrwxr-x 1 7095091 7500002 702878193 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/2007.csv*
-rwxrwxr-x 1 7095091 7500002 234052199 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/2008.csv*


In [10]:
# this syntax is used to look at all of them removing the year.
ls -la /anvil/projects/tdm/data/flights/subset/*.csv

-rwxrwxr-x 1 7095091 7500002 127162942 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/1987.csv*
-rwxrwxr-x 1 7095091 7500002 501039472 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/1988.csv*
-rwxrwxr-x 1 7095091 7500002 486518821 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/1989.csv*
-rwxrwxr-x 1 7095091 7500002 509194687 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/1990.csv*
-rwxrwxr-x 1 7095091 7500002 491210093 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/1991.csv*
-rwxrwxr-x 1 7095091 7500002 492313731 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/1992.csv*
-rwxrwxr-x 1 7095091 7500002 490753652 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/1993.csv*
-rwxrwxr-x 1 7095091 7500002 501558665 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/1994.csv*
-rwxrwxr-x 1 7095091 7500002 530751568 Jun 23  2020 /anvil/projects/tdm/data/flights/subset/1995.csv*
-rwxrwxr-x 1 7095091 7500002 533922363 Jun 23  2020 /anvil/projects/tdm/data/fligh

In [13]:
import pandas as pd
from pathlib import Path

Considering the files in this directory, which years of data are available? 1987 - 2008 data are available

What would the path be to access only the 2003 file in this data set?
ls /anvil/projects/tdm/data/flights/subset/2003.csv

for more information about the files
ls -la /anvil/projects/tdm/data/flights/subset/2003.csv

Go ahead and import the library pandas as pd, and import Path from pathlib

## Question 2

In [ ]:
# Notice that, in a range in Python, the final number in the range is not included.
# This code uses list comprehension to create a list of file paths, 
# ranging from the years of 1987-2008 files will now 
# contain strings of file paths for all the csv files in this directory.

In [14]:
files = [Path(f'/anvil/projects/tdm/data/flights/subset/{year}.csv') for year in range(1987,2009)]

In [15]:
files

[PosixPath('/anvil/projects/tdm/data/flights/subset/1987.csv'),
 PosixPath('/anvil/projects/tdm/data/flights/subset/1988.csv'),
 PosixPath('/anvil/projects/tdm/data/flights/subset/1989.csv'),
 PosixPath('/anvil/projects/tdm/data/flights/subset/1990.csv'),
 PosixPath('/anvil/projects/tdm/data/flights/subset/1991.csv'),
 PosixPath('/anvil/projects/tdm/data/flights/subset/1992.csv'),
 PosixPath('/anvil/projects/tdm/data/flights/subset/1993.csv'),
 PosixPath('/anvil/projects/tdm/data/flights/subset/1994.csv'),
 PosixPath('/anvil/projects/tdm/data/flights/subset/1995.csv'),
 PosixPath('/anvil/projects/tdm/data/flights/subset/1996.csv'),
 PosixPath('/anvil/projects/tdm/data/flights/subset/1997.csv'),
 PosixPath('/anvil/projects/tdm/data/flights/subset/1998.csv'),
 PosixPath('/anvil/projects/tdm/data/flights/subset/1999.csv'),
 PosixPath('/anvil/projects/tdm/data/flights/subset/2000.csv'),
 PosixPath('/anvil/projects/tdm/data/flights/subset/2001.csv'),
 PosixPath('/anvil/projects/tdm/data/fli

In [16]:
files[0]

PosixPath('/anvil/projects/tdm/data/flights/subset/1987.csv')

In [17]:
eightyseven = pd.read_csv(files[0]) 

In [19]:
print(eightyseven.columns)

Index(['Year', 'Month', 'DayofMonth', 'DayOfWeek', 'DepTime', 'CRSDepTime',
       'ArrTime', 'CRSArrTime', 'UniqueCarrier', 'FlightNum', 'TailNum',
       'ActualElapsedTime', 'CRSElapsedTime', 'AirTime', 'ArrDelay',
       'DepDelay', 'Origin', 'Dest', 'Distance', 'TaxiIn', 'TaxiOut',
       'Cancelled', 'CancellationCode', 'Diverted', 'CarrierDelay',
       'WeatherDelay', 'NASDelay', 'SecurityDelay', 'LateAircraftDelay'],
      dtype='object')


In [20]:
eightyseven.shape

(1311826, 29)

In [ ]:
# i think that we have flight data from Oct 1987 through April 2008
# i think that each year from 1988 to 2007 has all 12 months
# but the first and the last files are just a portion of the year.

In [26]:
# When you run this code it displayed the hidden column
pd.options.display.max_columns = None

In [27]:
pd.options.display.max_columns = None

In [28]:
eightyseven.head()

Year  Month  DayofMonth  DayOfWeek  DepTime  CRSDepTime  ArrTime  \
0  1987     10          14          3    741.0         730    912.0   
1  1987     10          15          4    729.0         730    903.0   
2  1987     10          17          6    741.0         730    918.0   
3  1987     10          18          7    729.0         730    847.0   
4  1987     10          19          1    749.0         730    922.0   

   CRSArrTime UniqueCarrier  FlightNum  TailNum  ActualElapsedTime  \
0         849            PS       1451      NaN               91.0   
1         849            PS       1451      NaN               94.0   
2         849            PS       1451      NaN               97.0   
3         849            PS       1451      NaN               78.0   
4         849            PS       1451      NaN               93.0   

   CRSElapsedTime  AirTime  ArrDelay  DepDelay Origin Dest  Distance  TaxiIn  \
0              79      NaN      23.0      11.0    SAN  SFO     447.0     NaN   
1              79      NaN      14.0      -1.0    SAN  SFO     447.0     NaN   
2              79      NaN      29.0      11.0    SAN  SFO     447.0     NaN   
3              79      NaN      -2.0      -1.0    SAN  SFO     447.0     NaN   
4              79      NaN      33.0      19.0    SAN  SFO     447.0     NaN   

   TaxiOut  Cancelled  CancellationCode  Diverted  CarrierDelay  WeatherDelay  \
0      NaN          0               NaN         0           NaN           NaN   
1      NaN          0               NaN         0           NaN           NaN   
2      NaN          0               NaN         0           NaN           NaN   
3      NaN          0               NaN         0           NaN           NaN   
4      NaN          0               NaN         0           NaN           NaN   

   NASDelay  SecurityDelay  LateAircraftDelay  
0       NaN            NaN                NaN  
1       NaN            NaN                NaN  
2       NaN            NaN                NaN  
3       NaN            NaN                NaN  
4       NaN            NaN                NaN

In [29]:
eightyseven.tail()

Year  Month  DayofMonth  DayOfWeek  DepTime  CRSDepTime  ArrTime  \
1311821  1987     12          11          5   1530.0        1530   1825.0   
1311822  1987     12          13          7   1530.0        1530   1815.0   
1311823  1987     12          14          1   1530.0        1530   1807.0   
1311824  1987     12           1          2   1525.0        1525   1643.0   
1311825  1987     12           2          3   1540.0        1525   1706.0   

         CRSArrTime UniqueCarrier  FlightNum  TailNum  ActualElapsedTime  \
1311821        1823            CO        638      NaN              115.0   
1311822        1823            CO        638      NaN              105.0   
1311823        1823            CO        638      NaN               97.0   
1311824        1638            CO        639      NaN               78.0   
1311825        1638            CO        639      NaN               86.0   

         CRSElapsedTime  AirTime  ArrDelay  DepDelay Origin Dest  Distance  \
1311821             113      NaN       2.0       0.0    ORD  EWR     719.0   
1311822             113      NaN      -8.0       0.0    ORD  EWR     719.0   
1311823             113      NaN     -16.0       0.0    ORD  EWR     719.0   
1311824              73      NaN       5.0       0.0    BOS  EWR     200.0   
1311825              73      NaN      28.0      15.0    BOS  EWR     200.0   

         TaxiIn  TaxiOut  Cancelled  CancellationCode  Diverted  CarrierDelay  \
1311821     NaN      NaN          0               NaN         0           NaN   
1311822     NaN      NaN          0               NaN         0           NaN   
1311823     NaN      NaN          0               NaN         0           NaN   
1311824     NaN      NaN          0               NaN         0           NaN   
1311825     NaN      NaN          0               NaN         0           NaN   

         WeatherDelay  NASDelay  SecurityDelay  LateAircraftDelay  
1311821           NaN       NaN            NaN                NaN  
1311822           NaN       NaN            NaN                NaN  
1311823           NaN       NaN            NaN                NaN  
1311824           NaN       NaN            NaN                NaN  
1311825           NaN       NaN            NaN                NaN

How many columns are there? 29 columns are there

What are the column names?

'Year', 'Month', 'DayofMonth', 'DayOfWeek', 'DepTime', 'CRSDepTime',
       'ArrTime', 'CRSArrTime', 'UniqueCarrier', 'FlightNum', 'TailNum',
       'ActualElapsedTime', 'CRSElapsedTime', 'AirTime', 'ArrDelay',
       'DepDelay', 'Origin', 'Dest', 'Distance', 'TaxiIn', 'TaxiOut',
       'Cancelled', 'CancellationCode', 'Diverted', 'CarrierDelay',
       'WeatherDelay', 'NASDelay', 'SecurityDelay', 'LateAircraftDelay'

Display the data from the first five rows.

## Question 3

In [24]:
# from row 0 to row 1311825 it has three letters (SAN ORD, BOS and so on)
eightyseven['Origin']

0          SAN
1          SAN
2          SAN
3          SAN
4          SAN
          ... 
1311821    ORD
1311822    ORD
1311823    ORD
1311824    BOS
1311825    BOS
Name: Origin, Length: 1311826, dtype: object

In [30]:
# This will show all the origin where the flight will take off from
eightyseven['Origin'].value_counts()

ORD    67216
ATL    66309
DFW    51860
LAX    45646
DEN    43376
       ...  
FOE       57
YAP       50
GUC       29
ILG       29
PIR        1
Name: Origin, Length: 237, dtype: int64

In [34]:
# for unique/specific
eightyseven['Origin'].value_counts()['ORD']

67216

In [36]:
eightyseven['Origin'].value_counts()['ATL']

66309

In [37]:
eightyseven['Origin'].value_counts()['DFW']

51860

In [38]:
# for unique/specific
eightyseven['Origin'].value_counts()['IND']

8817

In [ ]:
# This extract all of the counts of the number of times that each airport
# is the Origin city for a flight, i.e, where the flight departs.
# Then we extract the value for the indianapolis, e.g, in this case
# Indianapolis was the origin city for 8817 flights between Oct 1987 and Dec 1987.

Print out the unique elements of the Origin column.

Find the number of times that IND occurs in the Origin column. 8817 times that IND occurs in the Origin column

## Question 4

In [40]:
files[1]

PosixPath('/anvil/projects/tdm/data/flights/subset/1988.csv')

In [41]:
eightyeight = pd.read_csv(files[1]) 

In [42]:
eightyeight.shape

(5202096, 29)

In [44]:
pd.options.display.max_columns = None

In [43]:
eightyeight.head()

Year  Month  DayofMonth  DayOfWeek  DepTime  CRSDepTime  ArrTime  \
0  1988      1           9          6   1348.0        1331   1458.0   
1  1988      1          10          7   1334.0        1331   1443.0   
2  1988      1          11          1   1446.0        1331   1553.0   
3  1988      1          12          2   1334.0        1331   1438.0   
4  1988      1          13          3   1341.0        1331   1503.0   

   CRSArrTime UniqueCarrier  FlightNum  TailNum  ActualElapsedTime  \
0        1435            PI        942      NaN               70.0   
1        1435            PI        942      NaN               69.0   
2        1435            PI        942      NaN               67.0   
3        1435            PI        942      NaN               64.0   
4        1435            PI        942      NaN               82.0   

   CRSElapsedTime  AirTime  ArrDelay  DepDelay Origin Dest  Distance  TaxiIn  \
0              64      NaN      23.0      17.0    SYR  BWI     273.0     NaN   
1              64      NaN       8.0       3.0    SYR  BWI     273.0     NaN   
2              64      NaN      78.0      75.0    SYR  BWI     273.0     NaN   
3              64      NaN       3.0       3.0    SYR  BWI     273.0     NaN   
4              64      NaN      28.0      10.0    SYR  BWI     273.0     NaN   

   TaxiOut  Cancelled  CancellationCode  Diverted  CarrierDelay  WeatherDelay  \
0      NaN          0               NaN         0           NaN           NaN   
1      NaN          0               NaN         0           NaN           NaN   
2      NaN          0               NaN         0           NaN           NaN   
3      NaN          0               NaN         0           NaN           NaN   
4      NaN          0               NaN         0           NaN           NaN   

   NASDelay  SecurityDelay  LateAircraftDelay  
0       NaN            NaN                NaN  
1       NaN            NaN                NaN  
2       NaN            NaN                NaN  
3       NaN            NaN                NaN  
4       NaN            NaN                NaN

In [45]:
eightyeight.tail()

Year  Month  DayofMonth  DayOfWeek  DepTime  CRSDepTime  ArrTime  \
5202091  1988     12          12          1   1325.0        1325   2043.0   
5202092  1988     12          13          2   1325.0        1325   2038.0   
5202093  1988     12          14          3   1325.0        1325   2045.0   
5202094  1988     12           1          4   2027.0        2027   2152.0   
5202095  1988     12           2          5   2106.0        2027   2229.0   

         CRSArrTime UniqueCarrier  FlightNum  TailNum  ActualElapsedTime  \
5202091        2034            DL        162      NaN              318.0   
5202092        2034            DL        162      NaN              313.0   
5202093        2034            DL        162      NaN              320.0   
5202094        2145            DL        163      NaN               85.0   
5202095        2145            DL        163      NaN               83.0   

         CRSElapsedTime  AirTime  ArrDelay  DepDelay Origin Dest  Distance  \
5202091             309      NaN       9.0       0.0    HNL  LAX    2556.0   
5202092             309      NaN       4.0       0.0    HNL  LAX    2556.0   
5202093             309      NaN      11.0       0.0    HNL  LAX    2556.0   
5202094              78      NaN       7.0       0.0    ATL  MCO     403.0   
5202095              78      NaN      44.0      39.0    ATL  MCO     403.0   

         TaxiIn  TaxiOut  Cancelled  CancellationCode  Diverted  CarrierDelay  \
5202091     NaN      NaN          0               NaN         0           NaN   
5202092     NaN      NaN          0               NaN         0           NaN   
5202093     NaN      NaN          0               NaN         0           NaN   
5202094     NaN      NaN          0               NaN         0           NaN   
5202095     NaN      NaN          0               NaN         0           NaN   

         WeatherDelay  NASDelay  SecurityDelay  LateAircraftDelay  
5202091           NaN       NaN            NaN                NaN  
5202092           NaN       NaN            NaN                NaN  
5202093           NaN       NaN            NaN                NaN  
5202094           NaN       NaN            NaN                NaN  
5202095           NaN       NaN            NaN                NaN

In [46]:
eightyeight['Origin'].value_counts()

ORD    271494
ATL    259731
DFW    216849
LAX    169696
DEN    161146
        ...  
GST        86
RDD        19
ACV        19
ROP         1
BFI         1
Name: Origin, Length: 238, dtype: int64

In [47]:
eightyeight['Origin'].value_counts()['IND']

37399

In [48]:
files[2]

PosixPath('/anvil/projects/tdm/data/flights/subset/1989.csv')

In [49]:
eightynine = pd.read_csv(files[2]) 

In [50]:
eightynine.shape

(5041200, 29)

In [51]:
pd.options.display.max_columns = None

In [52]:
eightynine.head()

Year  Month  DayofMonth  DayOfWeek  DepTime  CRSDepTime  ArrTime  \
0  1989      1          23          1   1419.0        1230   1742.0   
1  1989      1          24          2   1255.0        1230   1612.0   
2  1989      1          25          3   1230.0        1230   1533.0   
3  1989      1          26          4   1230.0        1230   1523.0   
4  1989      1          27          5   1232.0        1230   1513.0   

   CRSArrTime UniqueCarrier  FlightNum  TailNum  ActualElapsedTime  \
0        1552            UA        183      NaN              323.0   
1        1552            UA        183      NaN              317.0   
2        1552            UA        183      NaN              303.0   
3        1552            UA        183      NaN              293.0   
4        1552            UA        183      NaN              281.0   

   CRSElapsedTime  AirTime  ArrDelay  DepDelay Origin Dest  Distance  TaxiIn  \
0             322      NaN     110.0     109.0    SFO  HNL    2398.0     NaN   
1             322      NaN      20.0      25.0    SFO  HNL    2398.0     NaN   
2             322      NaN     -19.0       0.0    SFO  HNL    2398.0     NaN   
3             322      NaN     -29.0       0.0    SFO  HNL    2398.0     NaN   
4             322      NaN     -39.0       2.0    SFO  HNL    2398.0     NaN   

   TaxiOut  Cancelled  CancellationCode  Diverted  CarrierDelay  WeatherDelay  \
0      NaN          0               NaN         0           NaN           NaN   
1      NaN          0               NaN         0           NaN           NaN   
2      NaN          0               NaN         0           NaN           NaN   
3      NaN          0               NaN         0           NaN           NaN   
4      NaN          0               NaN         0           NaN           NaN   

   NASDelay  SecurityDelay  LateAircraftDelay  
0       NaN            NaN                NaN  
1       NaN            NaN                NaN  
2       NaN            NaN                NaN  
3       NaN            NaN                NaN  
4       NaN            NaN                NaN

In [53]:
eightynine['Origin'].value_counts()

ORD    261259
DFW    228360
ATL    224036
LAX    163487
DEN    138407
        ...  
ROP        39
PFN        15
SUN        15
EGE        11
ABI         2
Name: Origin, Length: 237, dtype: int64

In [54]:
eightynine['Origin'].value_counts()['IND']

40567

In [55]:
files[3]

PosixPath('/anvil/projects/tdm/data/flights/subset/1990.csv')

In [56]:
ninety = pd.read_csv(files[3]) 

In [57]:
ninety.shape

(5270893, 29)

In [58]:
pd.options.display.max_columns = None

In [59]:
ninety.head()

Year  Month  DayofMonth  DayOfWeek  DepTime  CRSDepTime  ArrTime  \
0  1990      1           3          3   1707.0        1630   1755.0   
1  1990      1           4          4   1706.0        1630   1807.0   
2  1990      1           5          5   1629.0        1630   1715.0   
3  1990      1           6          6   1633.0        1630   1718.0   
4  1990      1           8          1   1630.0        1630   1726.0   

   CRSArrTime UniqueCarrier  FlightNum  TailNum  ActualElapsedTime  \
0        1723            US         29      NaN               48.0   
1        1723            US         29      NaN               61.0   
2        1723            US         29      NaN               46.0   
3        1723            US         29      NaN               45.0   
4        1723            US         29      NaN               56.0   

   CRSElapsedTime  AirTime  ArrDelay  DepDelay Origin Dest  Distance  TaxiIn  \
0              53      NaN      32.0      37.0    CMH  IND     182.0     NaN   
1              53      NaN      44.0      36.0    CMH  IND     182.0     NaN   
2              53      NaN      -8.0      -1.0    CMH  IND     182.0     NaN   
3              53      NaN      -5.0       3.0    CMH  IND     182.0     NaN   
4              53      NaN       3.0       0.0    CMH  IND     182.0     NaN   

   TaxiOut  Cancelled  CancellationCode  Diverted  CarrierDelay  WeatherDelay  \
0      NaN          0               NaN         0           NaN           NaN   
1      NaN          0               NaN         0           NaN           NaN   
2      NaN          0               NaN         0           NaN           NaN   
3      NaN          0               NaN         0           NaN           NaN   
4      NaN          0               NaN         0           NaN           NaN   

   NASDelay  SecurityDelay  LateAircraftDelay  
0       NaN            NaN                NaN  
1       NaN            NaN                NaN  
2       NaN            NaN                NaN  
3       NaN            NaN                NaN  
4       NaN            NaN                NaN

In [60]:
ninety['Origin'].value_counts()['IND']

43826

How many times is 'IND' the Origin airport in eightyeight? 37399 times was 'IND' the Origin airport in eightyeight

How many times is 'IND' the Origin airport in eightynine? 40567 times was 'IND' the Origin airport in eightynine

How many times is 'IND' the Origin airport in ninety? 43826 times was 'IND' the Origin airport in ninety


## Question 5

In [2]:
import pandas as pd
from pathlib import Path

In [3]:
files = [Path(f'/anvil/projects/tdm/data/flights/subset/{year}.csv') for year in range(1987,2009)]

In [4]:
pd.options.display.max_columns = None

In [5]:
# This is good reasoning and it would work except 
# our files are large and our kernel will fail.
count = 0
for file in files:
    df = pd.read_csv(file)
    count += len(df[df['Origin'] == 'IND'])

print(count)

/tmp/ipykernel_17/3180785423.py:3: DtypeWarning: Columns (22) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(file)


796496


In [6]:
# This method will work but, it will take a long time

total_count = 0
for file in files:
    for df in pd.read_csv(file, chunksize=10000):
        for index, row in df.iterrows():
            if row['Origin'] == 'IND':
                total_count += 1

print(total_count)

796496


In [ ]:
# this is a similar idea, but without the read_csv
# So we will not be storing the entire contents of any of the files in memory
# Instead, we will just loop through the file, one line at a time,
# checking to see if the Origin on that line of the file (the 16th field of the line)
# is equal to 'IND', and if it is, we add one to our counter called origin_ind
# At the very end we will print out the value of the counter, and that is equal to
# the total number of times that Indianapolis is the Origin city for a flight
# i.e, where a flight depart.

In [7]:
origin_ind = 0
for file in files:
    with open(file,'r') as f:
        for line in f:
            if line.split(",")[16] == 'IND':
                origin_ind += 1
print(origin_ind)

796496


Markdown notes and sentences and analysis written here.

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.